lib/bakeAmbientOcclusion.js

```diff
@@ -434,26 +437,34 @@ function computeAmbientOcclusionAt(position, normal, numberSamples, sqrtNumberSa
     for (var j = 0; j < numberSamples; j++) {
         var sampleRay = generateJitteredRay(position, normal, j, sqrtNumberSamples);
         aoBuffer.count[aoBufferIndex]++;
-        var nearestIntersect = naiveRaytrace(triangles, sampleRay, nearCull);
+        var nearestIntersect = uniformGridRaytrace(triangles, sampleRay, nearCull);
 
         if (nearestIntersect < rayDepth) {
             aoBuffer.samples[aoBufferIndex] += 1.0;
         }
     }
 }
 
-function naiveRaytrace(triangleSoup, ray, nearCull) {
-    // Check ray against every triangle in the soup. return the nearest intersection.
-    var minIntersect = Number.POSITIVE_INFINITY;
-    var triangleCount = triangleSoup.length;
-    for (var triangleSoupIndex = 0; triangleSoupIndex < triangleCount; triangleSoupIndex++) {
-        var positions = triangleSoup[triangleSoupIndex].positions;
-        var distance = rayTriangle(ray, positions[0], positions[1], positions[2], false);
-        if (defined(distance) && distance > nearCull) {
-            minIntersect = Math.min(distance, minIntersect);
-        }
+function raytraceNeighborFunction(positions, parameters) {
+    var distance = rayTriangle(parameters.ray, positions[0], positions[1], positions[2], false);
+    if (defined(distance) && distance > parameters.nearCull) {
+        parameters.minIntersect = Math.min(distance, parameters.minIntersect);
```

Comment:
I just realized: when looking in a grid cell, we don't even need to look at every single triangle. Anything this does after detecting a `distance` less than the max ray distance is wasted!